docs(openid-connect): session library v4.0.0 (#5115)
Kong/docs.konghq.com: app/_hub/kong-inc/openid-connect/_index.md

        - `off`: Do not set the Same-Site flag.
    - name: authorization_cookie_same_site
      minimum_version: "3.2.x"
      required: false
      default: '"Default"'
      datatype: string
      description: |
        Controls whether a cookie is sent with cross-origin requests, providing some protection against cross-site request forgery attacks:
        - `Strict`: Cookies will only be sent in a first-party context and not be sent along with requests initiated by third party websites.
        - `Lax`: Cookies are not sent on normal cross-site subrequests (for example to load images or frames into a third party site), but are sent when a user is navigating to the origin site (for instance, when following a link).
        - `None`: Cookies will be sent in all contexts, for example in responses to both first-party and cross-origin requests. If `SameSite=None` is set, the cookie Secure attribute must also be set (or the cookie will be blocked).
        - `Default`: Do not explicitly specify a Same-Site attribute.
    - name: authorization_cookie_httponly
      maximum_version: "3.1.x"
      required: false
      default: true
      datatype: boolean

>       description: 'Forbids JavaScript from accessing the cookie, for example, through the `Document.cookie` property.'

Reviewer: ```suggestion
      description: | 
        Forbids JavaScript from accessing the cookie, for example, through the `Document.cookie` property.
```

        - `Lax`: Cookies are not sent on normal cross-site subrequests (for example to load images or frames into a third party site), but are sent when a user is navigating to the origin site (for example, when following a link).
        - `None`: Cookies will be sent in all contexts, for example in responses to both first-party and cross-origin requests. If SameSite=None is set, the cookie Secure attribute must also be set (or the cookie will be blocked)
        - `off`: Do not set the Same-Site flag.
    - name: session_cookie_same_site
      minimum_version: "3.2.x"
      required: false
      default: '"Lax"'
      datatype: string
      description: |
        Controls whether a cookie is sent with cross-origin requests, providing some protection against cross-site request forgery attacks:
        - `Strict`: Cookies will only be sent in a first-party context and aren't sent along with requests initiated by third party websites.
        - `Lax`: Cookies are not sent on normal cross-site subrequests (for example to load images or frames into a third party site), but are sent when a user is navigating to the origin site (for example, when following a link).

>         - `None`: Cookies will be sent in all contexts, for example in responses to both first party and cross-origin requests. If SameSite=None is set, the cookie Secure attribute must also be set (or the cookie will be blocked)

Reviewer: ```suggestion
        - `None`: Cookies will be sent in all contexts, for example in responses to both first party and cross-origin requests. If `SameSite=None` is set, the cookie `Secure` attribute must also be set, or the cookie will be blocked.
```

>         - `Default`: Do not explicitly specify a Same-Site attribute.

Reviewer: ```suggestion
        - `Default`: Do not explicitly specify a `Same-Site` attribute.
```
Is it Same-Site or SameSite? The previous bullet point has it as SameSite.

Author: fixed it, thanks

    - name: session_cookie_httponly
      maximum_version: "3.1.x"
      required: false
      default: true
      datatype: boolean

>       description: 'Forbids JavaScript from accessing the cookie, for example, through the `Document.cookie` property.'

Reviewer: ```suggestion
      description: |
        Forbids JavaScript from accessing the cookie, for example, through the `Document.cookie` property.
```

        - `default`:  reuses session identifiers over modifications (but can be problematic with single-page applications with a lot of concurrent asynchronous requests)
        - `regenerate`: generates a new session identifier on each modification and does not use expiry for signature verification (useful in single-page applications or SPAs)
    - name: session_compressor
      maximum_version: "3.1.x"
      required: false
      default: '"none"'
      datatype: string
      description: |
        The session strategy:
        - `none`: no compression
        - `zlib`: use zlib to compress cookie data
    - name: session_audience
      minimum_version: "3.2.x"
      required: false
      default: '"default"'
      datatype: string

>       description: The session audience, for example "my-application"

Reviewer: ```suggestion
      description: The session audience, which is the intended target application. For example `"my-application"`.
```

    - name: session_remember
      minimum_version: "3.2.x"
      required: false
      default: false
      datatype: boolean

>       description: Enables or disables persistent sessions

Reviewer: ```suggestion
      description: Enables or disables persistent sessions.
```

    - name: session_remember_cookie_name
      minimum_version: "3.2.x"
      required: false
      default: '"remember"'
      datatype: string

>       description: Persistent session cookie name

Reviewer: ```suggestion
      description: Persistent session cookie name. Use with the `remember` configuration parameter.
```

    - name: session_remember_rolling_timeout
      minimum_version: "3.2.x"
      required: false
      default: 604800
      datatype: integer

>       description: Persistent session rolling timeout in seconds.

Reviewer: ```suggestion
      description: The persistent session rolling timeout window, in seconds.
```

    - name: session_remember_absolute_timeout
      minimum_version: "3.2.x"
      required: false
      default: 2592000
      datatype: integer

>       description: Persistent session absolute timeout in seconds.

Reviewer: ```suggestion
      description: The persistent session absolute timeout limit, in seconds.
```

    - name: session_request_headers
      minimum_version: "3.2.x"
      required: false
      default: null
      datatype: array of string elements
      description: |

>         List of information to include (as headers) in the request to upstream. 
>         Accepted values are: `id`, `audience`, `subject`, `timeout`, `idling-timeout`, `rolling-timeout`,
>         `absolute-timeout`.
>         For example, { "id", "timeout" } will set both `Session-Id` and `Session-Timeout` in the request headers.

Reviewer: ```suggestion
        List of information to include, as headers, in the request to the upstream. 
        Accepted values are: `id`, `audience`, `subject`, `timeout`, `idling-timeout`, `rolling-timeout`, and
        `absolute-timeout`.
        For example, `{ "id", "timeout" }` sets both `Session-Id` and `Session-Timeout` in the request headers.
```

    - name: session_response_headers
      minimum_version: "3.2.x"
      required: false
      default: null
      datatype: array of string elements
      description: |

>         List of information to include (as headers) in the response to downstream. 
>         Accepted values are: `id`, `audience`, `subject`, `timeout`, `idling-timeout`, `rolling-timeout`,
>         `absolute-timeout`.
>         For example, { "id", "timeout" } will inject both `Session-Id` and `Session-Timeout` in the response headers.

Reviewer: ```suggestion
        List of information to include, as headers, in the response to the downstream. 
        Accepted values are: `id`, `audience`, `subject`, `timeout`, `idling-timeout`, `rolling-timeout`, and
        `absolute-timeout`.
        For example: `{ "id", "timeout" }` injects both `Session-Id` and `Session-Timeout` in the response headers.
```

    - name: session_store_metadata
      minimum_version: "3.2.x"
      required: false
      default: false
      datatype: boolean
      description: | 
        Configures whether or not session metadata should be stored.

>         This includes information about the active sessions for a specific audience

Reviewer: ```suggestion
        This metadata includes information about the active sessions for a specific audience
```
